src/lib/CameraDetection.ts

```diff
@@ -0,0 +1,59 @@
+export const CameraDetection = (() => {
+  let canEnumerate = false;
+
+  if (navigator.mediaDevices && navigator.mediaDevices.enumerateDevices) {
+    navigator.enumerateDevices = async (callback) => {
+      const enumerateDevices = navigator.mediaDevices.enumerateDevices();
+      if (enumerateDevices && !!enumerateDevices.then) {
+        await navigator.mediaDevices.enumerateDevices().then(callback).catch(() => {
+          callback([]);
+        });
+      } else {
+        callback([]);
+      }
+    };
+  }
+
+  if (typeof MediaStreamTrack !== 'undefined' && 'getSources' in MediaStreamTrack) {
+    canEnumerate = true;
+  } else if (navigator.mediaDevices && !!navigator.mediaDevices.enumerateDevices) {
+    canEnumerate = true;
+  }
```

Comment:
I don't see reason to have 2 separate if blocks to do the same assignments. I suggest to merge them.